pkg/trace/propagation.go

```diff
@@ -138,16 +91,6 @@ func (t *otelTracer) contextFromHeaders(ctx context.Context, hdrs map[string][]s
 		ctx = ForceTracing(ctx)
 	}
 
-	if header.Get(OtelPropagationHeader) == "" {
-		prop, err := propagation.UnmarshalHoneycombTraceContext(header.Get(propagation.TracePropagationHTTPHeader))
-		if err == nil {
-			_, headers := propagation.MarshalW3CTraceContext(ctx, prop)
-			for k, v := range headers {
-				header.Set(k, v)
-			}
-		}
```

Comment:
This is the problem.

This block of code notices that there is no OpenTelemetry/W3C trace headers, then goes looking for Honeycomb headers.  If it finds them, it "converts" whatever it finds to W3C headers then hacks those re-exported values onto the request context so the true OpenTelemetry header parsing code further up the stack can use it.

The problem is that the honeycomb headers don't have a "sampled" flag like OpenTelemetry headers do.  They don't support parent-based sampling, or at least not in this way.  The resulting OpenTelemetry headers have the sampled flag hard-coded to "false", which means we never sample any of the traces based on requests that take this branch.